File: 240726/가장 짧은 부분합/shortest-subtotal.js
const main = () => {
    const input = require('fs').readFileSync(0).toString().trim().split("\n");

    const [n, s] = input[0].trim().split(" ").map(Number);
    const nums = input[1].trim().split(" ").map(Number);

    console.log(search(n, s, nums));
}

const search = (n, s, nums) => {
    let sum = 0;
    let l = 0;
    let r = 0;
    for (let i = 0; i < n; i++) {
        sum += nums[i];
        if (sum >= s) {
            r = i;
            break;
        }
    }

    if (sum < s) return -1;
    min = r - l + 1;

    while (r < n) {
        while (sum - nums[l] >= s) {
            sum -= nums[l];
            l++;
        }
        min = Math.min(min, r - l + 1);

        r++;
        sum += nums[r];
    }

    return min;
}

main();

File: 240728/가장 짧은 부분합/shortest-subtotal.kt
import java.io.*
import kotlin.math.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val nums =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
            .toTypedArray()
    val arr =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
            .toTypedArray()
    var min = Int.MAX_VALUE
    var p1 = 0
    var p2 = 0
    var total = arr[0]
    var count = 1
    while (p1 < nums[0] && p2 < nums[0]) {
        if (total >= nums[1]) {
            min = min(min, count)
            total -= arr[p1]
            p1++
            count--
        }
        if (p1 == p2 || total < nums[1]) {
            p2++
            if (p2 == nums[0]) {
                break
            }
            total += arr[p2]
            count++
        }
    }
    println(if (min == Int.MAX_VALUE) -1 else min)
}